internal/pkg/service/stream/storage/node/readernode/sliceupload/operator_test.go

```diff
@@ -0,0 +1,371 @@
+	"go.etcd.io/etcd/client/v3/concurrency"
+
+	commonDeps "github.com/keboola/keboola-as-code/internal/pkg/service/common/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/duration"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/rollback"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/config"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies"
+	bridgeTest "github.com/keboola/keboola-as-code/internal/pkg/service/stream/keboolasink/bridge/test"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/diskwriter/network/router/closesync"
+	volume "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/volume/model"
+	stagingConfig "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/staging/config"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/node/coordinator/slicerotation"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/node/readernode/sliceupload"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/test"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/test/testconfig"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
+)
+
+func TestSliceUpload(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	uploadTrigger := stagingConfig.UploadTrigger{
+		Count:    50000,
+		Size:     5 * datasize.MB,
+		Interval: duration.From(5 * time.Minute),
+	}
+
+	// The interval triggers upload conditions check
+	conditionsCheckInterval := time.Second
+
+	nodeID := "my-volume"
+	volumesPath := t.TempDir()
+	volumePath1 := filepath.Join(volumesPath, "hdd", "001")
+	require.NoError(t, os.MkdirAll(volumePath1, 0o700))
+	require.NoError(t, os.WriteFile(filepath.Join(volumePath1, volume.IDFile), []byte(fmt.Sprintf("%s-1", nodeID)), 0o600))
+	volumePath2 := filepath.Join(volumesPath, "hdd", "002")
+	require.NoError(t, os.MkdirAll(volumePath2, 0o700))
+	require.NoError(t, os.WriteFile(filepath.Join(volumePath2, volume.IDFile), []byte(fmt.Sprintf("%s-2", nodeID)), 0o600))
+	ignoredEtcdKeys := etcdhelper.WithIgnoredKeyPattern("^(definition)|(runtime)|(storage/file)|(storage/slice)|(storage/stats)|(storage/volume)")
+
+	// Create dependencies
+	clk := clock.NewMock()
+	clk.Set(utctime.MustParse("2000-01-01T00:00:00.000Z").Time())
+	deps, mock := dependencies.NewMockedCoordinatorScopeWithConfig(t, func(cfg *config.Config) {
```

Comment:
Tests depends on `Coordinator.SliceRotation` operator. The slices should be mocked instead. This is sort of E2E test that Coordinator -> ReaderNode integration works.